### ISSUE TITLE
Add support for folder mounted Windows disks

### ISSUE THREAD

**Issue by cruscio** [shirou/gopsutil#1245]
It appears disk/windows_disk.go filters on [drive letters](https://github.com/shirou/gopsutil/blob/e0ec1b9cda4470db704a862282a396986d7e930c/disk/disk_windows.go#L83), excluding folder mounted volumes. 

As a [Telegraf](https://github.com/influxdata/telegraf) user, I am looking to monitor *all* disk volumes mounted in Windows, both drive-letter mounted and folder-path mounted. Internally, Telegraf uses gopsutil to retrieve these data. Gopsutil appears to only support drive letters mounted disks.

**Additional context**

* Closed Telegraf bug: https://github.com/influxdata/telegraf/issues/8332
  * Expected behavior: When a volume is mounted to a folder in Windows, the `[[inputs.disk]]` plugin should output a series for that volume, with the `path` tag being the path to the folder mount point, and the `device` tag being either the volume DeviceId or the Disk & Partition numbers. 
  * Actual Behavior: Only volumes mounted to drive letters are returned

* To help reproduce this issue without needing a physical device, one can create a small VHD and mount it to a folder. Some potentially helpful links:
  * [Locating Mount Points Using PowerShell | Learn Powershell | Achieve More](https://learn-powershell.net/2012/08/10/locating-mount-points-using-powershell/)
  * [Mount all .vhdx files in a temporary folder : PowerShell](https://old.reddit.com/r/PowerShell/comments/chlo9d/mount_all_vhdx_files_in_a_temporary_folder/)
  * [Profiling disk mount points with Windows Perfmon | Convergence of Data, Cloud, and Infrastructure](https://www.davidklee.net/2012/06/17/profiling-disk-mount-points-with-windows-perfmon/)
  * [powershell - Correlate Physical Device ID to Volume Device ID - Stack Overflow](https://stackoverflow.com/questions/35878166/correlate-physical-device-id-to-volume-device-id)
  * [QEMU disk image utility — QEMU 6.2.50 documentation](https://qemu.readthedocs.io/en/latest/tools/qemu-img.html)

**Comment by rohan-rayaraddi** [comment]
@dloucasfx Is this being actively worked on? It would be great to have it fixed.

**Comment by dloucasfx** [comment]
> @dloucasfx Is this being actively worked on? It would be great to have it fixed.

@rohan-rayaraddi yes, I just updated the PR with @shirou 's suggestions https://github.com/shirou/gopsutil/pull/1534
